lib/gc/Transforms/GPU/AllocsToSLM.cpp

```diff
@@ -0,0 +1,107 @@
+//===- AllocsToSLM.cpp - A pass adding shared mem-space attr ----*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "gc/Transforms/Passes.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/GPU/TransformOps/Utils.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/IR/Dialect.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#include <numeric>
+#include <optional>
+
+using namespace mlir;
+using namespace mlir::gc;
+
+namespace mlir {
+namespace gc {
+#define GEN_PASS_DEF_ALLOCSTOSLM
+#include "gc/Transforms/Passes.h.inc"
+} // namespace gc
+} // namespace mlir
+
+namespace {
+
+bool isInGpuLaunch(mlir::Operation *op) {
+  // Traverse up through parent operations
+  mlir::Operation *parentOp = op;
+  while (parentOp) {
+    // Check if the current parent is a gpu.launch operation
+    if (llvm::isa<mlir::gpu::LaunchOp>(parentOp)) {
+      return true;
+    }
+    // Move to the parent operation
+    parentOp = parentOp->getParentOp();
+  }
```

Comment:
You're probably looking for `getParentOfType` here. Another option would be to just target the pass at gpu module instead. The latter is probably also useful if you want to check for the total allocated memory size without additional analysis.

Reply:
> Another option would be to just target the pass at gpu module instead

Targeting the pass to a gpu module would be nice, however it won't work with the existing pipeline. There are some passes that must be run after `allocs-to-slm` but before conversion to the gpu module (`linalg-to-xegpu`, `linalg-to-loops`, `insert-gpu-allocs`). 

Some of the passes (`linalg-to-xegpu` and `linalg-to-loops`) might be adjusted by retargeting them from `funcOp` to `gpuFuncOp`, but `insert-gpu-allocs` won't work that way, as its logic expects to be applied to a `func.func` containing a `gpu.launch block` (and not a GPU module).

Reply:
Sorry, I was thinking about something else while typing. I meant the launch gpu op, not gpu module. Anyway, the change is trivial.